pip/flatpak-pip-generator

```diff
@@ -158,22 +161,36 @@ def fprint(string: str) -> None:
 
 packages = []
 if opts.requirements_file:
-    requirements_file = os.path.expanduser(opts.requirements_file)
+    requirements_file_input = os.path.expanduser(opts.requirements_file)
     try:
-        with open(requirements_file, 'r') as req_file:
+        with open(requirements_file_input, 'r') as req_file:
             reqs = parse_continuation_lines(req_file)
             reqs_as_str = '\n'.join([r.split('--hash')[0] for r in reqs])
-            packages = list(requirements.parse(reqs_as_str))
+            reqs_list_raw = reqs_as_str.splitlines()
+            py_version_regex = re.compile(r';.*python_version .+$') # Remove when pip-generator can handle python_version
+            reqs_list = [py_version_regex.sub('', p) for p in reqs_list_raw]
+            if opts.ignore_pkg:
+                reqs_new_list = [i for i in reqs_list if i not in opts.ignore_pkg]
+                reqs_new = '\n'.join(reqs_new_list)
+            else:
+                reqs_new = reqs_as_str
+            packages = list(requirements.parse(reqs_new))
+            with tempfile.NamedTemporaryFile('w', delete=False, prefix='requirements.') as req_file:
```

Comment:
This just leaves temp files around. Move the scope up and let python delete it when the script exits.

Reply:
Global variable or moving it up - neither works. I'm not exactly sure why.

I can add it as a global variable so that it doesn't need repeating for the `elif opts.packages` part, if you want.

What works is `os.remove` -ing it after `subprocess.run` is done. Right now, it is only called if an error occurs. `

Reply:
Oh I missed that this was already done elsewhere.

It is fine for now I suppose.

Reply:
Handled now